azure-auth-helper/src/main/java/com/microsoft/azure/auth/AzureCredentialCallable.java

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.auth;
+
+import com.microsoft.aad.adal4j.AuthenticationContext;
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+public class AzureCredentialCallable implements Callable<AzureCredential> {
+    private String baseUrl;
+    private AcquireTokenFunction acquireTokenFunc;
+
+    public AzureCredentialCallable(String baseUrl, AcquireTokenFunction acquireTokenFunc) {
+        if (baseUrl == null) {
+            throw new NullPointerException("Parameter 'baseUrl' cannot be null.");
```

Comment:
This should be IllebalArguemntExcpetion. NPE is more a runtime exception.

Reply:
funny discussions at https://stackoverflow.com/questions/3881/illegalargumentexception-or-nullpointerexception-for-a-null-parameter, will change to IAE